iree/compiler/Codegen/SPIRV/AdrenoConfig.cpp

```diff
@@ -231,10 +125,14 @@ LogicalResult setAdrenoCodeGenConfig(const spirv::TargetEnv &,
   return TypeSwitch<Operation *, LogicalResult>(rootOp)
       .Case<linalg::BatchMatmulOp, linalg::MatmulOp>(
           [](auto op) { return setOpConfig(op); })
-      .Case<linalg::Conv2DNhwcHwcfOp>(
-          [](auto op) { return setOpConfig(op, 32); })
-      .Case<linalg::DepthwiseConv2DNhwOp>(
-          [](auto op) { return setOpConfig(op, 16); })
+      .Case<linalg::Conv2DNhwcHwcfOp>([](auto op) {
+        return setConvOpConfig(op, /*subgroupSize=*/64,
```

Comment:
Is there a reason to hardwire this. If this is based on some device properties, we could derive a value from that.

Reply:
Good catch. We should get the subgroup size from the target env. Changed that.